### PR TITLE
PHP7  - use spaceship

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: tests/Console/Report/ListSetsReport/JsonReporterTest.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ protected function getFormat()
     protected function assertFormat($expected, $input): void
     {
         static::assertJsonSchema($input);
-        static::assertJsonStringEqualsJsonString($expected, $input);
+        static::assertSame($expected, $input);
```
Comment (keradus) [inline tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: why the change? comparing `assertJsonStringEqualsJsonString` over `assertSame` is nice, as `assertJsonStringEqualsJsonString` will ignore JSON-irrelevant whitespaces

Comment (SpacePossum) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: `assertSame` is more strict, as `assertJsonStringEqualsJsonString` normalizes hashes so it doesn't test if elements are in order that is expected

Comment (keradus) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: order of element in json hash should not be relevant. why you think it is relevant for us?
(remember it's JSON, not PHP assoc array that is keeping the order of keys)

Comment (SpacePossum) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: I assumed it matter because of the sorting here:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.1/src/Console/Report/ListSetsReport/JsonReporter.php#L42
which I was changes and wanted to test

Comment (keradus) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: the assoc-array we have may be having sorted items, but after we convert to json with `json_encode`, we have no control whether the JSON is having sorted keys or not.
some implementations are explicitly aiming that, eg https://www.npmjs.com/package/json-stable-stringify for node, but I'm not aware about any promise like that in PHP.

then, whoever reads JSON generated by us, they may not keep the order we produced anyway, eg if read by perl, the order of keys in hash is undefined.

Comment (keradus) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: (also, `assertJsonStringEqualsJsonString` should have better error message than `assertSame` when showing diff for 2 JSONs)

Comment (SpacePossum) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php:41]: ok, should I remove the sorting from report builders as these hold no value? would be nice clean up :)

Comment (keradus) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php]: sorting in builder was helping me debugging whether it generates content properly, I would keep it tbh ;)

Comment (SpacePossum) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php]: that gets us in the odd spot that I cannot fix it because you don't like the fix to be tested ;)

Comment (SpacePossum) [reply tests/Console/Report/ListSetsReport/JsonReporterTest.php]: restored, the output generators don't have to sort it seems as the source already sorts:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.1/src/RuleSet/RuleSets.php#L45
but I left as is for now

File: src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php [hunk 2]
```diff
@@ -300,16 +300,10 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
      */
     protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
     {
-        $allowed = self::$tokenMap;
-        $allowed['yield_from'] = true; // TODO remove this when update to PHP7.0
-        ksort($allowed);
```
Comment (keradus) [inline src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php:305]: maybe we should still sort the keys we are passing to `AllowedValueSubset` ?

Comment (SpacePossum) [reply src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php:305]: the token map is already sorted so no need to sort it again I think?

Comment (keradus) [reply src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php:305]: only by coincidence, right? nothing is ensuring anymore that values we gonna show as possible options are sorted
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5838/files#diff-e62493a3eac53bf195d7e4e5ba28f97cafe0c59da890c623c2b52082008a2fa9R41
(eg, we can add new value to that structure at end of it and not in alpha order)

Comment (SpacePossum) [reply src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php:305]: if I add a utest for that would be enough or are there more thing here that need fixing or should we drop the PR?

Comment (SpacePossum) [reply src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php:305]: >only by coincidence, right? 
no, I typed it in that order

**Comment by keradus** [comment]
Thank you @SpacePossum.
